src/compiler/checker.ts

```diff
@@ -33873,6 +33895,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         // If candidate is undefined, it means that no candidates had a suitable arity. In that case,
         // skip the checkApplicableSignature check.
         if (reportErrors) {
+            if (!headMessage && isCallExpression(node) && isSyntheticHasInstanceMethodCall(node)) {

```

Comment:
This all seems like complexity that follows from the synthetic call business. Do we even need to do call and overload resolution in the first place?

Reply:
It seems to me that we should, otherwise we would be drastically inconsistent with how the transposed representation works, as I mentioned in [another comment](https://github.com/microsoft/TypeScript/pull/55052#discussion_r1333852020).

Reply:
We also use the synthetic expression mechanism to properly type check tagged templates, decorators, and spread arguments where synthesizing a transposed representation makes type checking far easier than one off mechanisms.

Reply:
The synthetic call is actually far less complicated than the mechanism we currently employ for "call-like expressions" like tagged templates, decorators, and Jsx elements, which depend on threading those expressions through `getResolvedSignature`, `resolveSignature`, and their own independent `resolveTaggedTemplateExpression`/`resolveDecorator`/`resolveJsxOpeningLikeElement` functions, as well as all of the complexity behind `getEffectiveCallArguments` et al. In the long term, it might actually be cleaner for us to generate synthetic `CallExpression` nodes for those expressions instead, and remove all of the complicated branching we do there.

Reply:
This has been switched over to use the `getResolvedSignature` logic.